packages/office-ui-fabric-react/src/components/Coachmark/Coachmark.base.tsx

```diff
@@ -376,19 +394,30 @@ export class CoachmarkBase extends BaseComponent<ICoachmarkProps, ICoachmarkStat
     const { preventDismissOnLostFocus } = this.props;
     const currentDoc: Document = getDocument()!;
 
-    this._events.off();
-
     if (currentDoc) {
-      this._events.on(currentDoc, 'keydown', this._onKeyDown, true);
-
+      this._async.setTimeout(() => {
```

Comment:
What's the timeout for?

Reply:
I followed the example in [CalloutContent](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Callout/CalloutContent.base.tsx#L328) but it looks like there the delay seems required in order to prevent React from firing async focus events in IE when the target changes focus quickly before rendering the callout which I don't think is a problem in the case of Coachmark - looked further and Popup & FocusZone also don't use timeouts for disposables.push (on ()) cases.

Reply:
Removed setTimeout wrappings from the disposables.push(on()) cases in add & remove methods.